app/code/Magento/Inventory/Model/SourceCarrierLink.php

```diff
@@ -30,39 +30,40 @@ protected function _construct()
     /**
      * @inheritdoc
      */
-    public function getCarrierCode()
+    public function getCarrierCode(): ?string
     {
         return $this->getData(self::CARRIER_CODE);
     }
 
     /**
      * @inheritdoc
      */
-    public function setCarrierCode($carrierCode)
+    public function setCarrierCode(?string $carrierCode): void
     {
         $this->setData(self::CARRIER_CODE, $carrierCode);
     }
 
     /**
      * @inheritdoc
      */
-    public function getPosition()
+    public function getPosition(): ?int
     {
-        return $this->getData(self::POSITION);
+        $position = $this->getData(self::POSITION);
+        return $position !== null ? (int)$position : null;
     }
 
     /**
      * @inheritdoc
      */
-    public function setPosition($position)
+    public function setPosition(?int $position): void
     {
         $this->setData(self::POSITION, $position);
```

Comment:
Need to check behavior
Because if we fill data based on form submitting position will be `string`